### DISCUSSION TITLE
How do I apply migrations?

### DISCUSSION THREAD

**Discussion by flowirtz** [OWASP/OpenCRE#160] [Q&A] [answered]
I followed the installation instructions in the README and I can successfully start the app.
The db migrations don't seem to get applied though, so no data can be fetched in the FE.

Here are the steps I followed:
```bash
cp cres/db.sqlite standards_cache.sqlite
make install
make dev-run
```

The app starts fine. But when I click on a standard from the footer of the homepage, eg `/standard/OWASP%20Top%2010%202021` I get a "This page does not exist" error page.
Similarly, when I select `/cre/482-866`, I get an `Document could not be loaded` in the FE.

Looking at the server logs, I think the issue lies here: `sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: cre_links`. And indeed, inspecting the `standards_cache.sqlite`, the table is called `crelinks` and not `cre_links`. I assume that's because the migration `3c65127871a6_change_standards_table_to_nodes` didn't get applied.

I tried applying the migration via `make migrate-upgrade`, but that results in the following error:
```bash
$ make migrate-upgrade

[ -d "./venv" ] && . ./venv/bin/activate
export FLASK_APP=/Users/ping/Desktop/Git/common-requirement-enumeration/cre.py
flask db upgrade
Usage: flask db upgrade [OPTIONS] [REVISION]

Error: Could not import "cre".
make: *** [migrate-upgrade] Error 2
```

Any suggestions on how I can apply the migrations?
Thanks!

**Comment by northdpole** [comment, marked as answer]
There was a bug on make migrations it is now fixed, can you please try again?

**Reply by flowirtz** [reply]
It still didn't work for me on Mac, however, I think I fixed it now:

I changed the `FLASK_APP` in the `Makefile`:

```bash
migrate-upgrade:
	[ -d "./venv" ] && . ./venv/bin/activate
	export FLASK_APP=cre
	flask db upgrade  
```

Now applying the migrations works for me.


I think you're on Windows, right? Can you maybe try that as well?
If it works I'm happy to put up a PR for it.

Thanks!
